### PR TITLE
Version 1.3.2

### PR DESCRIPTION
**Pull request by pFernbach**
## Main changes
* Change the architecture of the folders. Add a python/mlp folder containing all the python files.
* Add jrl-cmakemodule and CMakeLists file to install the package
* Refactor `run_mlp.py` to a class (`loco_planner.py`) and a `__main__.py`  script
* Change the way the configuration files are used: `config.py` now define a class that must be instanciated. It should be imported and instanciated only once, it is then forwarded to all the methods requiring it. It may be modified by the user before / between the functions call. 

## Other changes

* Update relative to https://github.com/humanoid-path-planner/hpp-rbprm-corba/pull/61 in rbprm
* Add a 'simulator' block. The only method for now is the basic integration with pinocchio as previously done in the tsid script. 
This block is used to update the state of the robot inside the wholebody scripts
* Update to changes in gepetto-viewer loadObstacleModel API
* Fix tsid output when stopping before the end of the sequence
* Clean debug outputs
* Always compute ZMP for '_REF' contact sequence file
* fix numerical approximation issue in momentumopt wrapper

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ This package rely on a lot of optionnal packages, see the section 'Available met
 ## Install depencies from binary repositories :
 
 1. Add robotpkg to your apt configuration: http://robotpkg.openrobots.org/debian.html
-2. `sudo apt update && sudo apt install robotpkg-py27-pinocchio robotpkg-py27-multicontact-api`
+2. `sudo apt update && sudo apt install robotpkg-py35-pinocchio robotpkg-py35-multicontact-api`
```
Comment (nim65s) [inline README.md]: `py3\*`, otherwise this will work only on 16.04 :)

File: python/mlp/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+from .loco_planner import LocoPlanner
+from .config import Config
```
Comment (nim65s) [inline python/mlp/__init__.py]: missing new line

File: python/mlp/centroidal/none.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from  mlp.utils.requirements import Requirements as CentroidalInputsNone
+from  mlp.utils.requirements import Requirements as CentroidalOutputsNone
+
+def generate_centroidal_none(cfg, cs, cs_initGuess=None, fullBody=None, viewer=None, first_iter = True):
+    print("Centroidal trajectory not computed !")
+    return None
```
Comment (nim65s) [inline python/mlp/centroidal/none.py]: `return None` is equivalent to no return

File: python/mlp/end_effector/bezier_predef.py [hunk 4]
```diff
@@ -1,13 +1,22 @@
...
 import numpy.linalg
 import numpy as np
 from curves import bezier, piecewise_bezier, SE3Curve, piecewise_SE3
 import hpp_bezier_com_traj as bezier_com
 import math
+from mlp.utils.requirements import Requirements
+from mlp.utils.cs_tools import generate_effector_trajectories_for_sequence
 pinocchio.switchToNumpyArray()
 
+class EffectorInputsBezier(Requirements):
```
Comment (nim65s) [inline python/mlp/end_effector/bezier_predef.py:13]: style:  2 blank lines before a class definition

File: python/mlp/end_effector/smoothed_foot.py [hunk 5]
```diff
@@ -23,4 +26,7 @@ def generateEndEffectorTraj(time_interval,
     return SmoothedFootTrajectory(time_interval, [placement_init, placement_end])
 
 def effectorCanRetry():
-    return False
\ No newline at end of file
+    return False
+
+def generate_effector_trajectories_for_sequence_smoothed_foot(cfg, cs,  fullBody = None):
+    return generate_effector_trajectories_for_sequence(cfg, cs, generateEndEffectorTraj, fullBody)
```
Comment (nim65s) [inline python/mlp/end_effector/smoothed_foot.py]: missing line return

**Comment by nim65s** [comment]
Congrats :)

I only have really minor remarks :)

**Comment by pFernbach** [comment]
Update to work without the viewer and unit-tests will be in another PR.

**Pull request merged**